### PR TITLE
Register with phone number

### PR DESCRIPTION
**Pull request by NathanielBennett**
## What does this change?

We want to conditionally add a clientId - by means of a switch - that makes new users provide a phone-number when they register in order to comment. 
## What is the value of this and can you measure success?

We want to see if we can improve the quality of discussions by precluding RBUs ( repeat banned users ) from signing up ..
## Does this affect other platforms - Amp, Apps, etc?

N/A
## Screenshots

N/A But see here for the altered registration flow. https://github.com/guardian/identity-frontend/pull/157
## Request for comment

@piuccio @nicl

### CODE REVIEW

File: common/app/conf/switches/FeatureSwitches.scala [hunk 1]
```diff
@@ -506,4 +506,13 @@ trait FeatureSwitches {
     exposeClientSide = true
   )
 
+  val RegisterWithPhoneNumber = Switch(
```
Comment (piuccio) [inline common/app/conf/switches/FeatureSwitches.scala]: I'd put this switch in `DiscussionSwitches`

File: common/app/views/fragments/discussionFooter.scala.html [hunk 2]
```diff
@@ -8,8 +8,9 @@
         <div class="container__meta">
             <h2 class="container__meta__title">comments <span class="discussion__comment-count js-discussion-comment-count"></span></h2>
 
-            <p class="container__meta__item discussion__meta discussion__meta--open-signed-out"><a class="u-underline" href="@Configuration.id.url/signin?INTCMP=DOTCOM_COMMENTS_SIGNIN">Sign in</a>
-                or <a class="u-underline" href="@Configuration.id.url/register?INTCMP=DOTCOM_COMMENTS_REG">create your Guardian account</a> to join the discussion.
+            <p class="container__meta__item discussion__meta discussion__meta--open-signed-out">
+                <a class="u-underline" href="@Configuration.id.url/signin?INTCMP=DOTCOM_COMMENTS_SIGNIN@if(RegisterWithPhoneNumber.isSwitchedOn){&clientId=discussion"}>Sign in</a>
```
Comment (piuccio) [inline common/app/views/fragments/discussionFooter.scala.html]: the `clientId` in the other PR is `comments` not `discussion`

https://github.com/guardian/identity-frontend/pull/157/files#diff-852ecc2984d9c2bad5a4a7a160fde5a9R33

Comment (NathanielBennett) [inline common/app/views/fragments/discussionFooter.scala.html]: @piuccio  aw thanks, man. :(

**Comment by piuccio** [comment]
👍

**Pull request merged**